### ISSUE TITLE
Forward search after build stopped working on MacOS + Skim on 5.5.0

### ISSUE THREAD

**Issue by bvdmitri** [latex-lsp/texlab#880]
This is a follow-up on https://github.com/latex-lsp/texlab/issues/475

After the recent release the forward search after build stopped working properly with `Skim` application on MacOS. Previosuly everything was working fine, but now it basically jumps on the most top-level `\section` after each build, which may be quite far away from the actual cursor position.

I should note that manual forward search still works fine. The problem is only with the automatic forward search after build.

Can I somehow control the settings and restore the previous behaviour as it was working as expected? 
My setup is Neovim + LSP + Texlab and the pdf reader is `Skim`. 
My config is:
```lua
------------------- Latex LS settings -----------------
-- requires brew install --cask mactex & the Skim application
vim.g.texlabwd = string.format("/tmp/texlab-latex/%s", vim.g.wd) 
vim.g.texlab_latexindent_config = string.format("%s/latexindent.yaml", vim.g.pwd)

-- Create texlab working directory if it does not exist
os.execute(string.format("mkdir -p %s", vim.g.texlabwd))

lspconfig.texlab.setup {
  on_attach = on_attach,
  log_level = vim.lsp.protocol.MessageType.Log,
  -- cmd = { "texlab", "-vvvv", "--log-file=/tmp/texlab.log" },
  settings = {
    texlab = {
      auxDirectory = vim.g.texlabwd,
      build = {
        -- executable = "tectonic",
        -- args = {  "-X", "compile", "%f", "-p", "--synctex", "--keep-logs", "--keep-intermediates", "--outdir", vim.b.texlabwd },
        executable = "latexmk",
        args = { "-pdf", "-f", "-interaction=nonstopmode", "-synctex=1", string.format("-outdir=%s", vim.g.texlabwd), "%f" },
        onSave = true,
        forwardSearchAfter = true
      },
      chktex = {
        onOpenAndSave = true
      },
      forwardSearch = {
        executable = "/Applications/Skim.app/Contents/SharedSupport/displayline",
        args = { "-g", "%l", "%p", "%f" }
      },
      latexFormatter = "latexindent",
      latexindent = {
        ['local'] = vim.g.texlab_latexindent_config,
        modifyLineBreaks = true
      }
    }
  },
  capabilities = lsp_capabilities
}

```

**Comment by bvdmitri** [comment]
Ah, I understand now, basically if I first use the format command + build (in my setup I use `latexindent`) the position after build is wrong. If I simply save the file without formatting everything works as expected. But I usually do formatting + save and this is currently broken.

**Comment by pfoerster** [comment]
@bvdmitri Thanks for the report! The issue was that `texlab` was resetting the cursor position to zero after the change notification resulting from the formatting request was processed. This is fixed now by b61313e.

**Comment by pfoerster** [comment]
Released with `v5.5.1`

**Comment by bvdmitri** [comment]
Hey! Thanks, but as soon as I can tell the issue is still present on `5.5.1`.

**Comment by carlosala** [comment]
@pfoerster also happening for me, since v5.5.1 forward search is not working for me

**Comment by carlosala** [comment]
I tried and was working for me today. I needed to reinstall texlab and not sure what was going on but it worked after reinstalling. I'm sure I touched something in my way there xD

**Comment by carlosala** [comment]
What's not fully working for me it's inverse search though.

**Comment by pfoerster** [comment]
@bvdmitri  
Can you give #889 a try? It should now be much more reliable.

**Comment by pfoerster** [comment]
@carlosala

> What's not fully working for me it's inverse search though.

Inverse search is not done by the server, so you need to check the configuration of your PDF viewer.

**Comment by MarkusLohmayer** [comment]
Hey! I am on 5.7.0 and forward search after build does not work (always sends line number 1) while regular forward search works.

**Comment by pfoerster** [comment]
@MarkusLohmayer I have a few questions:

- Which LSP client are you using? 
- Does it also happen if you trigger the build manually or is it always after saving? 
- Do you have formatting on save enabled?

**Comment by MarkusLohmayer** [comment]
Dear @pfoerster, thanks for reopening.
- I am using NeoVim's native LSP client.
- If I use `:TexlabForward` everything works nicely. Only when I set `texlab.build.forwardSearchAfter` (obviously together with `texlab.build.onSave`) then I always got line number 1 when I tried last. Now I am trying again (re-enabling `texlab.build.forwardSearchAfter`), and I observe that I get the line number where I last made an edit, which is not necessarily the cursor position, but anyways much better.
- I am pretty sure I have no formatting on save enabled. Which option would that be? I don't see anything getting formatted anyways. I also have no auto-command calling `lua vim.lsp.buf.format()`.

Let me know if I can provide more information which would be useful.
